skills/hook-kit/resources/staged-protect.sh

```
@@ -0,0 +1,31 @@
+[ -z "$FILE_PATH" ] && exit 0
+
+# Pass through if not a git repository
+git rev-parse --git-dir >/dev/null 2>&1 || exit 0
+
+# Check the list of staged files
+STAGED_FILES=$(git diff --cached --name-only 2>/dev/null)
+[ -z "$STAGED_FILES" ] && exit 0
+
+# Convert the absolute path to a relative path
+REPO_ROOT=$(git rev-parse --show-toplevel 2>/dev/null)
+REL_PATH="${FILE_PATH#$REPO_ROOT/}"
```

Shellcheck (0.11.0)

[info] Expansions inside ${..} need to be quoted separately, otherwise they match as patterns.

(SC2295)